### PR TITLE
Text colors

### LINKED ISSUE #776: [Enhancement] Add Font-Colors (White, Black, Semi-Dark to start with)
**Is your enhancement request related to a problem? Please describe.**
We had some a11y findings which involved our semi-dark color, which was used in "unselected" states on both text and components. 

The findings are based upon that (interactive)components inactive states should also have 3:1 contrast to be AA. 

https://zpl.io/2jQB6XQ

Skærmbillede 2020-01-28 kl  11 36 52

**Describe the solution you'd like**
1. Rename semi-dark to semi-dark-text and change its color to #707070
2. Add semi-dark-graphic with a color #8e8e8e
3. Change component "unselected" states (eg. radio/toggle/checkmark) to semi-dark-graphic

Toggle: https://zpl.io/VKQ3JPW
Checkmark: https://zpl.io/2GRqAZm

### PR DESCRIPTION
**Pull request by MortenGregersen**
Closes #776

- [x] Docs have been added / updated (for bug fixes / features)

## PR Type

What kind of change does this PR introduce?

- [x] Bugfix
- [x] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [x] Documentation content changes
- [ ] cookbook application / infrastructure changes
- [ ] Other... Please describe:

## What is the current behavior?

Issue Number: #776

## What is the new behavior?

Kirby now has text colors and a `get-text-color` SCSS function. All components has been updated to use the new function.

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

File: src/kirby/scss/_global-styles.scss [hunk 1]
```diff
@@ -8,6 +8,9 @@
   @each $color, $color-name in $kirby-colors {
     --kirby-#{$color}: #{$color-name};
   }
+  @each $color, $color-name in $text-colors {
+    --kirby-text-#{$color}: #{$color-name};
```
Comment (jakobe) [inline src/kirby/scss/_global-styles.scss]: Include `color` in variable name:
```suggestion
    --kirby-text-color-#{$color}: #{$color-name};
```

File: src/kirby/scss/base/_functions.scss [hunk 2]
```diff
@@ -44,6 +44,11 @@
   }
 }
 
+@function get-text-color($variant) {
+  $value: _get-map-prop($text-colors, $variant);
+  @return unquote('var(--kirby-text-' + $variant + ')');
```
Comment (jakobe) [inline src/kirby/scss/base/_functions.scss]: Include `color` in variable name:
```suggestion
  @return unquote('var(--kirby-text-color-' + $variant + ')');
```

File: src/kirby/components/chip/chip.component.scss [hunk 3]
```diff
@@ -14,34 +14,34 @@ $chip-height: 30px;
   margin-right: 5px;
   cursor: pointer;
   background-color: get-color('white');
-  color: get-color('black');
+  color: get-text-color('black');
```
Comment (jakobe) [inline src/kirby/components/chip/chip.component.scss]: Use contrast instead;
```suggestion
  color: get-color('white-contrast');
```

File: src/kirby/components/chip/chip.component.scss [hunk 4]
```diff
@@ -14,34 +14,34 @@ $chip-height: 30px;
...
   user-select: none;
 
   &:hover,
   &.is-selected {
     background-color: get-color('black');
-    color: get-color('white');
+    color: get-text-color('white');
```
Comment (jakobe) [inline src/kirby/components/chip/chip.component.scss]: Use contrast instead;
```suggestion
  color: get-color('black-contrast');
```

File: src/kirby/components/chip/chip.component.scss [hunk 5]
```diff
@@ -14,34 +14,34 @@ $chip-height: 30px;
...
   }
 
   :host-context(.default),
   :host-context(.kirby-color-brightness-white) {
     background-color: get-color('white');
-    color: get-color('black');
+    color: get-text-color('black');
```
Comment (jakobe) [inline src/kirby/components/chip/chip.component.scss]: Use contrast instead;
```suggestion
  color: get-color('white-contrast');
```

File: src/kirby/components/chip/chip.component.scss [hunk 6]
```diff
@@ -14,34 +14,34 @@ $chip-height: 30px;
...
   :host-context(.kirby-color-brightness-dark) {
     background-color: transparent;
-    color: get-color('white');
+    color: get-text-color('white');
 
     &:hover,
     &.is-selected {
       background-color: get-color('white');
-      color: get-color('black');
+      color: get-text-color('black');
```
Comment (jakobe) [inline src/kirby/components/chip/chip.component.scss]: Use contrast instead;
```suggestion
  color: get-color('white-contrast');
```

File: src/kirby/components/chip/chip.component.scss [hunk 7]
```diff
@@ -14,34 +14,34 @@ $chip-height: 30px;
...
   :host-context(.default),
   :host-context(.kirby-color-brightness-white) {
     background-color: get-color('white');
-    color: get-color('black');
+    color: get-text-color('black');
     &:hover,
     &.is-selected {
       background-color: get-color('black');
-      color: get-color('white');
+      color: get-text-color('white');
```
Comment (jakobe) [inline src/kirby/components/chip/chip.component.scss]: Use contrast instead;
```suggestion
  color: get-color('black-contrast');
```

File: src/kirby/components/chip/chip.component.scss [hunk 8]
```diff
@@ -14,34 +14,34 @@ $chip-height: 30px;
...
     }
   }
 
   :host-context(.kirby-color-brightness-dark) {
     background-color: transparent;
-    color: get-color('white');
+    color: get-text-color('white');
```
Comment (jakobe) [inline src/kirby/components/chip/chip.component.scss]: Not sure about this one? 🤔

Comment (MortenGregersen) [reply src/kirby/components/chip/chip.component.scss]: I could be dark-contrast, but not sure if this is right.

File: src/app/showcase/fonts-showcase/fonts-showcase.component.scss [hunk 9]
```diff
@@ -2,7 +2,7 @@
 
 code {
   background-color: get-color('medium');
-  color: get-color('black');
+  color: get-text-color('black');
```
Comment (jakobe) [inline src/app/showcase/fonts-showcase/fonts-showcase.component.scss]:
```suggestion
  color: get-color('medium-contrast');
```

File: src/app/showcase/fonts-showcase/fonts-showcase.component.scss [hunk 10]
```diff
@@ -25,7 +25,7 @@ code {
   }
 
   code.highlight {
-    color: get-color('black');
+    color: get-text-color('black');
```
Comment (jakobe) [inline src/app/showcase/fonts-showcase/fonts-showcase.component.scss]: Obsolete - delete 🔥

File: src/kirby/components/form-field/_form-field-inputs.shared.scss [hunk 11]
```diff
@@ -10,7 +10,7 @@
   border-radius: size('s');
   box-shadow: get-elevation(2);
   box-sizing: border-box;
-  color: get-color('black');
+  color: get-text-color('black');
```
Comment (jakobe) [inline src/kirby/components/form-field/_form-field-inputs.shared.scss]: Consider moving up next to  ` background-color: get-color('white');` and sync:
```suggestion
  color: get-color('black-contrast');
```

File: src/kirby/components/form-field/_form-field-inputs.shared.scss [hunk 12]
```diff
@@ -6,11 +6,12 @@
 
 :host {
   background-color: get-color('white');
+  color: get-color('white-contrast');
   border: none;
   border-radius: size('s');
   box-shadow: get-elevation(2);
   box-sizing: border-box;
-  color: get-text-color('black');
+
```
Comment (jakobe) [inline src/kirby/components/form-field/_form-field-inputs.shared.scss]: :fire: Whitespace

**Pull request merged**